src/pages/SubmissionDetails.js

```diff
@@ -0,0 +1,234 @@
+        }, ['Submit a new workflow'])
+      ]),
+      div([
+        h(TextCell, [(h(Link, { onClick: () => Nav.goToPath('submission-history') }, ['Submission History'])), ' >', ` Submission ${submissionId}`])
+      ]),
+      div({
+        style: {
+          marginTop: '1em', height: tableHeight({ actualRows: paginatedPreviousRuns.length, maxRows: 12.5, heightPerRow: 250 }), minHeight: '10em'
+        }
+      }, [
+        h(AutoSizer, [
+          ({ width, height }) => h(FlexTable, {
+            'aria-label': 'previous runs',
+            width, height, sort,
+            rowCount: paginatedPreviousRuns.length,
+            noContentMessage: 'Nothing here yet! Your previously run workflows will be displayed here.',
+            hoverHighlight: true,
+            rowHeight,
+            rowWidth,
+            columns: [
+              {
+                size: { basis: 350 },
+                field: 'run_id',
+                headerRenderer: () => h(Sortable, { sort, field: 'run_id', onSort: setSort }, ['Run ID']),
+                cellRenderer: ({ rowIndex }) => {
+                  return h(TextCell, [paginatedPreviousRuns[rowIndex].run_id])
+                }
+              },
+              {
+                size: { basis: 220, grow: 0 },
+                field: 'state',
+                headerRenderer: () => h(Sortable, { sort, field: 'state', onSort: setSort }, ['Status']),
+                cellRenderer: ({ rowIndex }) => {
+                  const failureStates = ['SYSTEM_ERROR', 'EXECUTOR_ERROR']
+                  if (failureStates.includes(paginatedPreviousRuns[rowIndex].state)) {
+                    return div({ style: { width: '100%', textAlign: 'center' } }, [
+                      div({ style: { marginBottom: '1rem', fontWeight: 'bold' } }, [h(TextCell, {}, [icon('warning-standard', { size: 18, color: 'red' }), ['   Failed with error']])]),
+                      h(Link, { style: {}, onClick: () => setViewErrorsId(rowIndex) }, ['View'])
+                    ])
+                  } else { return h(TextCell, [paginatedPreviousRuns[rowIndex].state]) }
+                }
+              },
+              {
+                size: { basis: 300, grow: 0 },
+                field: 'submission_date',
+                headerRenderer: () => h(Sortable, { sort, field: 'submission_date', onSort: setSort }, ['Submission date']),
+                cellRenderer: ({ rowIndex }) => {
+                  return h(TextCell, [Utils.makeCompleteDate(paginatedPreviousRuns[rowIndex].submission_date)])
+                }
+              },
+              {
+                size: { basis: 300, grow: 0 },
+                field: 'duration',
+                headerRenderer: () => h(Sortable, { sort, field: 'duration', onSort: setSort }, ['Duration']),
+                cellRenderer: ({ rowIndex }) => {
+                  const terminalStates = ['COMPLETE', 'CANCELED', 'SYSTEM_ERROR', 'ABORTED', 'EXECUTOR_ERROR']
```

Comment:
Agreed. I'll move the version of this function in SubmissionHistory there for now. I think I'll leave it to Katrina to replace it here in SubmissionDetails, because I think she has some other changes in flight and it'll save her a conflict resolution step...